Cargo.toml

```
@@ -43,10 +43,13 @@ mpmc_large = []
 portable-atomic = { version = "1.0", optional = true }
 hash32 = "0.3.0"
 serde = { version = "1", optional = true, default-features = false }
-stable_deref_trait = { version = "1", default-features = false }
 ufmt-write = { version = "0.1", optional = true }
 defmt = { version = ">=0.2.0,<0.4", optional = true }
 
+# for the pool module
+[target.'cfg(any(target_arch = "arm", target_arch = "aarch64", target_arch = "x86"))'.dependencies]
```

```suggestion
[target.'cfg(any(target_arch = "arm", target_arch = "x86"))'.dependencies]
```

Sorry, I'm on a desktop now.  I mean that I think `aarch64` can be excluded here and it should still work.

---

I tried that, and it fails the build. Are you saying if I remove the `build.rs` stuff, those builds will succeed without the `aarch64` target?

I'll try.

---

Which build failed?

I tried this out on a branch (https://github.com/newAM/heapless/commit/42cc06d0631f334da998aed02508e6aa9205e97a) and the CI passed (https://github.com/newAM/heapless/actions/runs/6844625067)

It shouldn't need `build.rs` changes.

---

I'll try it again, but all the `arm` targets failed.

---

Literally no idea why those builds were failing, some weird heisenbug. The changes as they are now, are literally what I had in the first commit...

Anyway, builds pass now.